### DISCUSSION TITLE
Optimizing mul function in diyp

### DISCUSSION THREAD

**Discussion by wjr-z** [nlohmann/json#3782] [Ideas]
I noticed that the mul function in diyfp still uses a simple algorithm. In fact, simple inline assembly can be used in clang and GCC, and in MSVC__ UMULH for optimization.

**Comment by nlohmann** [comment]
Looking into it.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
